src/sdk/putResources.js

```diff
@@ -18,7 +18,8 @@ function makePutResources(concurrency = DEFAULT_CONCURRENCY_LIMIT) {
 
     const needMoreResources = runningRender.getNeedMoreResources();
     if (needMoreResources) {
-      for (const resource of rGridDom.getResources()) {
+      const availableReosurces = Array.from(new Set(allResources.concat(rGridDom.getResources())));
```

Comment:
why is this necessary? `allResources` should contain everything already

Reply:
For backwards compatibility. I didn't fix all the places that invoke this function